[GB.QT5]
* BUG: The component now should compile correctly with C++11.
Authored by:  gambas 2017-01-09
Browse code at this revision
Parent: [r8048]
Child: [r8050]
changed	/gambas/trunk/gb.qt4/src/CMenu.cpp
changed	/gambas/trunk/gb.qt4/src/CWidget.cpp
changed	/gambas/trunk/gb.qt5/src/Makefile.am
/gambas/trunk/gb.qt4/src/CMenu.cpp Diff Switch to side-by-side view
--- a/gambas/trunk/gb.qt4/src/CMenu.cpp
+++ b/gambas/trunk/gb.qt4/src/CMenu.cpp
@@ -59,7 +59,7 @@
 
 #define HANDLE_PROXY(_ob) \
 	while (EXT(_ob) && EXT(_ob)->proxy) \
-		_ob = (typeof _ob)(EXT(_ob)->proxy);
+		_ob = (__typeof__ _ob)(EXT(_ob)->proxy);
 
 static CMENU_EXT *alloc_ext(CMENU *_object)
 {
/gambas/trunk/gb.qt4/src/CWidget.cpp Diff Switch to side-by-side view
--- a/gambas/trunk/gb.qt4/src/CWidget.cpp
+++ b/gambas/trunk/gb.qt4/src/CWidget.cpp
@@ -96,7 +96,7 @@
 
 #define HANDLE_PROXY(_ob) \
 	while (EXT(_ob) && EXT(_ob)->proxy) \
-		_ob = (typeof _ob)(EXT(_ob)->proxy);
+		_ob = (__typeof__ _ob)(EXT(_ob)->proxy);
 
 static CWIDGET_EXT *alloc_ext(CWIDGET *_object)
 {
/gambas/trunk/gb.qt5/src/Makefile.am Diff Switch to side-by-side view
--- a/gambas/trunk/gb.qt5/src/Makefile.am
+++ b/gambas/trunk/gb.qt5/src/Makefile.am
@@ -8,7 +8,7 @@
 
 gb_qt5_la_LIBADD = @THREAD_LIB@ @QT5_LIB@
 gb_qt5_la_LDFLAGS = -module @LD_FLAGS@ @QT5_LDFLAGS@
-gb_qt5_la_CXXFLAGS = @THREAD_INC@ -DGB_QT_COMPONENT $(AM_CXXFLAGS)
+gb_qt5_la_CXXFLAGS = @THREAD_INC@ -DGB_QT_COMPONENT $(AM_CXXFLAGS) -std=c++11
 gb_qt5_la_CPPFLAGS = @QT5_INC@ -I$(top_srcdir)/share/
 
 gb_qt5_la_SOURCES = \
